docs/src/main/asciidoc/spanner.adoc

```diff
@@ -25,51 +27,180 @@ dependencies {
 }
 ----
 
+This setup takes care of bringing in the latest compatible version of Cloud Java Spanner libraries as well.
+
+== Spring Data Spanner
+
+
+=== Configuration
+
+To setup Spring Data Spanner, you have to configure connection to Cloud Spanner and Repositories (if you want to use them).
+
+==== Spanner settings
+
+You will need to use link:../spring-cloud-gcp-starters/spring-cloud-gcp-starter-data-spanner[Spring Boot Starter for Spring Data Spanner] to configure Google Cloud Spanner in your Spring application. It contains all the necessary setup that makes it easy to authenticate with your Google Cloud project.
+The following configuration options are available:
+
+|===
+| Name | Description | Optional | Default value
+| `spring.cloud.gcp.spanner.projectId` | the GCP project ID to use | Yes | provided by Spring Cloud GCP
+| `spring.cloud.gcp.spanner.instanceId` | Spanner instance to use | No |
+| `spring.cloud.gcp.spanner.database` |
+Spanner database to use | No |
+|===
+
+==== Repository settings
+
+Spring Data Repositories (see below for functionality) can be configured via the `@EnableSpannerRepositories` annotation on your main configuration file (typically the `@SpringBootApplication` file.
```

Comment:
I have a bit of a pet peeve with using parenthesis in general, and in specific, in a reference document that is meant to be polished. :)

Instead of "see below for functionality", add a link to it and remove the parenthesis.

Instead of (typically, start a new sentence: Typically, this is the file which contains the `@SpringBootApplication` file.

There is also no requirement that I know of that `@EnableSpannerRepositories` must be in the same file as the spring boot application. It should work in any `@Configuration` file.